src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
The following preprocessor-based solution might be another option to consider, the advantage being that it is independent of the cpuinfo module.
```
# File: TargetArch.cmake
#
# This uses the C preprocessor to determine if the target architecture is x86_64.
# Created based on TargetArch.cmake from solar-cmake.
#
# References:
# https://github.com/axr/solar-cmake
# http://stackoverflow.com/a/12024211

set(archdetect_c_code "
#if defined(__x86_64) || defined(__x86_64__) || defined(__amd64) || defined(__amd64__) || defined(_M_AMD64)  || defined(_M_X64)
    #error cmake_ARCH x86_64
#else
    #error cmake_ARCH other
#endif
")

function(target_arch_is_x86_64 output_var)
    file(WRITE "${CMAKE_BINARY_DIR}/arch.c" "${archdetect_c_code}")
    enable_language(C)
    try_run(
        run_result_unused
        compile_result_unused
        "${CMAKE_BINARY_DIR}"
        "${CMAKE_BINARY_DIR}/arch.c"
        COMPILE_OUTPUT_VARIABLE ARCH
    )
    string(REGEX MATCH "cmake_ARCH ([a-zA-Z0-9_]+)" ARCH "${ARCH}")
    string(REPLACE "cmake_ARCH " "" ARCH "${ARCH}")
    if(${ARCH} STREQUAL "x86_64")
        set(${output_var} 1 PARENT_SCOPE)
    else()
        set(${output_var} 0 PARENT_SCOPE)
    endif()
endfunction()
```
We could then use
```
include(TargetArch)
target_arch_is_x86_64(ARCH_IS_X86_64)
```
and use `WORKBENCH_USE_SIMD AND ARCH_IS_X86_64` in conditionals. If we wanted to use this, we would probably have to retain their copyright notice since what I created above is just a boiled down version of TargetArch.cmake from https://github.com/axr/solar-cmake.